observable_operator.go

```diff
@@ -1145,6 +1145,95 @@ func (op *ignoreElementsOperator) end(_ context.Context, _ chan<- Item) {
 func (op *ignoreElementsOperator) gatherNext(_ context.Context, _ Item, _ chan<- Item, _ operatorOptions) {
 }
 
+func (o *ObservableImpl) Join(joiner Func2, right Observable, ttExtractor func(interface{}) int64, window int64, opts ...Option) Observable {
```

Comment:
Durations should be based on `rxgo.Duration`. This way it can be tested using `timeCausality()` (if possible)

Reply:
Fixed